stat/roc.go

```diff
@@ -98,28 +96,35 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 		}
 		nPos += posWeight
 		nNeg += negWeight
-		if y[i] <= cutoffs[bin] {
+		if y[i] < cutoffs[bin] {
+			// Actually counting false negatives
 			tpr[bin] += posWeight
+			// Actually counting true negatives
 			fpr[bin] += negWeight
 		}
 	}
+	for bin < len(cutoffs)-1 {
+		bin++
+		tpr[bin] = tpr[bin-1]
+		fpr[bin] = fpr[bin-1]
+	}
 
 	invNeg := 1 / nNeg
 	invPos := 1 / nPos
 	for i := range tpr {
 		// Prevent fused float operations by
 		// making explicit float64 conversions.
+		// Convert negative counts to TPR and FPR
```

Comment:
I think this would be better above the for loop. Also add final full stop.

Reply:
Yep